core-bundle/src/Framework/ContaoFramework.php

```diff
@@ -185,7 +185,7 @@ private function setConstants(): void
         if ('FE' === $this->getMode() && ($session = $this->getSession()) && $this->request->hasPreviousSession()) {
             $session->start();
 
-            \define('BE_USER_LOGGED_IN', $this->tokenChecker->hasBackendUser() && $this->tokenChecker->isPreviewMode());
+            \define('BE_USER_LOGGED_IN', $this->tokenChecker->isPreviewMode());
```

Comment:
IMO this is not resolved yet because this change hast to be reverted in this pull request.

```suggestion
            \define('BE_USER_LOGGED_IN', $this->tokenChecker->hasBackendUser() && $this->tokenChecker->isPreviewMode());
```

Reply:
jup, this is just a draft atm 🙃

Reply:
I reviewed this again, and I think this change needs to stay. Otherwise all legacy extensions relying on `BE_USER_LOGGED_IN` will incorrectly not show stuff in the new preview link mode. Again, the constant was never meant to represent whether a back end user was actually logged in … 🤷

Reply:
The preview link mode is new (also doesn’t show a preview toolbar for example), so I think it is OK if it behaves different. For security reasons and to get extension developers to use the new method I would revert this.

Reply:
It makes the preview links likely unusable in my opinion …

Reply:
Yes, without this change the preview URL will not work with any extension that still uses `BE_USER_LOGGED_IN`. Imho it is fine to change this because there is not actual behaviour change anyway, is there? Except for the new feature.

Reply:
> there is no actual behaviour change anyway
Technically there is IMO. Because now `BE_USER_LOGGED_IN` in no longer secured to a backend login. For example if you force 2FA, the preview link still works without 2FA.

But I agree, this “issue” is most probably only a theoretical possibility. So I think we can keep it as it is currently in this pull request 👍

@fritzmg can/should we document this at least that A) `BE_USER_LOGGED_IN` in now deprecated and B) can also be true for the new preview links.

Reply:
Yes https://github.com/contao/docs/issues/923 :)